examples/hitl/rearrange_v2/ui.py

```diff
@@ -628,6 +601,101 @@ def _draw_pickable_object_highlights(self):
                     destination_mask=dest_mask,
                 )
 
+    def _draw_hovered_object_highlights(self) -> None:
+        """Highlight the hovered object."""
+        object_id = self._hover_selection.object_id
+        if object_id is None:
+            return
+
+        sim = self._sim
+        obj = sim_utilities.get_obj_from_id(
+            sim, object_id, self._world._link_id_to_ao_map
+        )
+        if obj is None:
+            return
+
+        # Draw the outline of the highlighted interactable link (e.g. drawer in a cabinet).
+        if self._is_object_interactable(object_id):
+            link_index = self._world.get_link_index(object_id)
+            if link_index:
+                link_node = obj.get_link_scene_node(link_index)
+                reachable = self._can_open_close_receptacle(
+                    link_node.translation
+                )
+                color = COLOR_VALID if reachable else COLOR_INVALID
+                color = self._to_color_array(color)
+                color[3] = 0.3  # Make hover color dimmer than selection.
+                self._gui_drawer._client_message_manager.draw_object_outline(
+                    priority=1,
+                    color=color,
+                    line_width=8.0,
+                    object_ids=[object_id],
+                    destination_mask=Mask.from_index(self._user_index),
+                )
+
+        # Skip highlight if select object is the same as hovered object.
+        selected_obj_id = self._click_selection.object_id
+        if selected_obj_id is not None:
+            selected_obj = sim_utilities.get_obj_from_id(
+                sim, selected_obj_id, self._world._link_id_to_ao_map
+            )
+            if selected_obj is not None and obj.handle == selected_obj.handle:
+                return
```

Comment:
Nothing happens after this exit check, do we need it? :thinking:

Reply:
That's indeed an old code block that went past the radar.
Thanks :)